### PR TITLE
test: add codeql scan

### LINKED ISSUE #8: Add CodeQL scan workflow
EDIT: I thought we aimed for 100% test coverage, but I don't know if we need that for now.
Jest says we're already at 100% coverage for lines/branches/statements.

I've re-jigged this issue to apply to the CodeQL scanner addition.

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes:

- Add boilerplate CodeQL scan (this is what GitHub suggests from the security tab)

## Context:

I've grabbed the CodeQL boilerplate that GitHub suggests (go to Security tab -> Code scanning alerts -> CodeQL Analysis).
It contains a few differences to the one we use over on the Renovate bot repo though, so we need to review it together and make changes if needed.

Closes #8

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ main ]
+  schedule:
+    - cron: '15 17 * * 5'
```
Comment (HonkingGoose) [inline .github/workflows/codeql-analysis.yml:21]:
```suggestion
    - cron: '0 13 * * 1'
```

This is what `renovatebot/renovate` uses.

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
```
Comment (HonkingGoose) [inline .github/workflows/codeql-analysis.yml:30]: Seems the newer CodeQL action file is clearer on the permissions it needs.

File: .github/workflows/codeql-analysis.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+    strategy:
+      fail-fast: false
```
Comment (HonkingGoose) [inline .github/workflows/codeql-analysis.yml:33]: `renovatebot/renovate` does not use the `fail-fast`: false strategy. I think we would only need this if we used more than one programming language in our project?

If you want I can remove it.

**Review by HonkingGoose — COMMENTED** [review]
Some more comments...

**Review by rarkins — APPROVED** [review]

**Pull request merged**